action/src/plugins/staticHosting/helpers.ts

```diff
@@ -168,8 +169,31 @@ export const uploadFileWithSignedUrl = async (
   }
 };
 
+const injectAnnotationPlugin = (localFilePath: string) => {
+  if (!localFilePath.toLowerCase().endsWith('.html')) {
```

Comment:
What if it's a directory that also has an index.html file in it?

Reply:
This helper only process files. It is called in two places: one for file and one for directories. When it is a directory, the caller will iterate through each file.

Reply:
The non-single page test coverage report is a directory with a group of html files.

Reply:
Cool.